### ISSUE TITLE
AcceptDialog did not show the whole title if the title is too long

### ISSUE THREAD

**Issue by jsjtxietian** [godotengine/godot#86287]
### Tested versions

Godot v4.3.dev (b94eb58d3)

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (NVIDIA; 31.0.15.3619) - 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz (16 Threads)

### Issue description

When trying to reproduce https://github.com/godotengine/godot/issues/86226, I noticed that the AcceptDialog with the title : "Auto Create Tiles in Non-Transparent Texture Regions?" does not shown the whole title as the image shown:

![image](https://github.com/godotengine/godot/assets/8315986/7182ab4e-1e78-4f63-b193-6f7d7f330cd2)

After some quick search, I find that in the ctor of AcceptDialog it did call the `set_keep_title_visible(true)`, but the window's window_id is -1 so the following responsible code did not get excuted.
```
if (!embedder && window_id != DisplayServer::INVALID_WINDOW_ID && keep_title_visible) {
	Size2i title_size = DisplayServer::get_singleton()->window_get_title_size(tr_title, window_id);
	size_limit = size_limit.max(title_size);
}
```

### Steps to reproduce

Follow the same step as https://github.com/godotengine/godot/issues/86226
After drag the provided png to create a tileset, the dialog will appear.

### Minimal reproduction project (MRP)

Same as https://github.com/godotengine/godot/issues/86226

**Comment by YuriSizov** [comment]
> After some quick search, I find that in the ctor of AcceptDialog it did call the `set_keep_title_visible(true)`, but the window's window_id is -1 so the following responsible code did not get excuted.

That's not why it doesn't get executed. We don't even call `_update_window_size()` from `set_keep_title_visible()` is the window ID is invalid. And whenever we assign the window ID, we do call `_update_window_size()`.

In your screenshot you're using the single-window mode. Which means that all windows are embedded. Which doesn't pass the `!embedder` check. This feature for keeping the title fully visible is only for detached windows, I'm not sure we have anything in place for embedded windows.

**Comment by jsjtxietian** [comment]
Yes the screen shot above is using single window mode, but if I change to not use this mod, same will happen too :

![image](https://github.com/godotengine/godot/assets/8315986/dfb4e359-d5a6-478b-a386-4b61ab029953)

Step into it will clearly show that window_id is -1 when the AcceptDialog pop up:

![image](https://github.com/godotengine/godot/assets/8315986/47e326fa-fc90-4b7a-a85e-b80e59c9f107)

**Comment by eugene87222** [comment]
Hi @jsjtxietian, does this issue still exist? I cannot reproduce this in Windows 11 Godot v4.3.dev (#9c626b6).

**Comment by jsjtxietian** [comment]
@eugene87222  I'm stilling can see this in Godot v4.3.dev (ae51db75e) - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (NVIDIA; 31.0.15.3619) - 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz (16 Threads)

![image](https://github.com/godotengine/godot/assets/8315986/b61654b3-57de-4d9b-81df-a1d008e4c037)

**Comment by TheSofox** [comment]
Not able to reproduce:

![image](https://github.com/user-attachments/assets/0afa02a4-ed76-4849-a4b4-957bdd6b0a5d)

Godot v4.4.dev (efc7c628b) - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1070 (NVIDIA; 31.0.15.3623) - Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz (8 Threads)
